storage/src/main/java/org/apache/kafka/storage/internals/log/RemoteStorageThreadPool.java

```diff
@@ -41,6 +44,19 @@ public String logPrefix() {
                 return "[" + Thread.currentThread().getName() + "]";
             }
         }.logger(RemoteStorageThreadPool.class);
+        KafkaMetricsGroup metricsGroup = new KafkaMetricsGroup(this.getClass());
+        metricsGroup.newGauge(metricsNamePrefix.concat("TaskQueueSize"), new Gauge<Integer>() {
+            @Override
+            public Integer value() {
+                return RemoteStorageThreadPool.this.getQueue().size();
```

Comment:
Is this call thread safe? Note that there may be two threads working with the queue data structure at the same time, the JMX thread populating this metric and this RLM thread.

Reply:
Do we need that kind of accuracy for this metric? A bit of staleness should be okay, right?

Reply:
More than accuracy, I was concerned about the impact of having two different threads accessing a non-thread safe data structure. In some cases, it could leave the structure in an inconsistent state. However, unlike complex structures like Map, seems like this queue's implementation is simple and size() just does a `AtomicInteger.get`. 

Hence, my concern is mitigated. Please consider this comment resolved.